cmd/panacead/cmd/encrypt_data.go

```diff
@@ -0,0 +1,106 @@
+package cmd
+
+import (
+	"encoding/hex"
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	"github.com/medibloc/panacea-core/v2/crypto"
+	oracletypes "github.com/medibloc/panacea-core/v2/x/oracle/types"
+	"github.com/spf13/cobra"
+	"github.com/tendermint/tendermint/libs/cli"
+	"os"
+)
+
+func EncryptDataCmd(defaultNodeHome string) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "encrypt-data [input-file-path] [output-file-path] [key-name]",
+		Short: "Encrypt data with shared key which consists of oracle public key and provider's private key",
+		Long: `
+			This command can encrypt data with shared key which consists of oracle public key and provider's private key.
+			The key to be used for encryption should be stored in the localStore.
+			If not stored, please add the key first via the following command.
+			panacead keys add ...
+		`,
+		Args: cobra.ExactArgs(3),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			queryClient := oracletypes.NewQueryClient(clientCtx)
+
+			origData, err := os.ReadFile(args[0])
+			if err != nil {
+				return err
+			}
+
+			params, err := queryClient.Params(cmd.Context(), &oracletypes.QueryOracleParamsRequest{})
+			if err != nil {
+				return err
+			}
+
+			oraclePubKey := params.GetParams().GetOraclePublicKey()
+
+			encryptedData, err := encrypt(clientCtx, args[2], origData, oraclePubKey)
+			if err != nil {
+				return err
+			}
+
+			if err := os.WriteFile(args[1], encryptedData, 0644); err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+
+	cmd.PersistentFlags().String(flags.FlagHome, defaultNodeHome, "The application home directory")
+	cmd.PersistentFlags().String(flags.FlagKeyringDir, "", "The client Keyring directory; if omitted, the default 'home' directory will be used")
+	cmd.PersistentFlags().String(flags.FlagKeyringBackend, flags.DefaultKeyringBackend, "Select keyring's backend (os|file|test)")
+	cmd.PersistentFlags().String(cli.OutputFlag, "text", "Output format (text|json)")
+	cmd.PersistentFlags().String(flags.FlagChainID, "", "The network chain ID")
+
+	flags.AddQueryFlagsToCmd(cmd)
+
+	return cmd
+}
+
+func encrypt(clientCtx client.Context, keyName string, origData []byte, oraclePubKeyStr string) ([]byte, error) {
+	// get unsafe export private key from keystore
+	privKeyHex, err := keyring.NewUnsafe(clientCtx.Keyring).UnsafeExportPrivKeyHex(keyName)
+	if err != nil {
+		return nil, err
+	}
+
+	privKeyBz, err := hex.DecodeString(privKeyHex)
+	if err != nil {
+		return nil, err
+	}
+
+	privKey, _ := btcec.PrivKeyFromBytes(btcec.S256(), privKeyBz)
+
+	// oracle public key
+	oraclePubKeyBz, err := hex.DecodeString(oraclePubKeyStr)
```

Comment:
Do we use `Hex` for encoding?
As far as I know, it's `Base64`, But I'm not sure :)

Reply:
fixed in [e7034f6](https://github.com/medibloc/panacea-core/pull/529/commits/e7034f6d80a9bd312c1ca93aaa848243c8309d5c)